config/initializers/10-patches.rb

```diff
@@ -73,39 +73,25 @@ def self.human_attribute_name(attr, options = {})
 
 module ActiveModel
   class Errors
-    def full_messages
-      full_messages = []
-
-      @messages.each_key do |attribute|
-        @messages[attribute].each do |message|
-          next unless message
-
-          if attribute == :base
-            full_messages << message
-          elsif attribute == :custom_values
-            # Replace the generic "custom values is invalid"
-            # with the errors on custom values
-            @base.custom_values.each do |value|
-              full_messages += value.errors.map do |_, message|
-                I18n.t(:"errors.format",
-                       default:   '%{attribute} %{message}',
-                       attribute: value.custom_field.name,
-                       message:   message
-                )
-              end
-            end
-          else
-            attr_name = attribute.to_s.gsub('.', '_').humanize
-            attr_name = @base.class.human_attribute_name(attribute, default: attr_name)
-            full_messages << I18n.t(:"errors.format",
-                                    default:   '%{attribute} %{message}',
-                                    attribute: attr_name,
-                                    message:   message
-            )
-          end
-        end
+    def full_message(attribute, message)
+      return message if attribute == :base
+
+      # if a model acts_as_customizable it will inject attributes like 'custom_field_1' into itself
+      # using attr_name_override we resolve names of such attributes.
+      # The rest of the method should reflect the original method implementation of ActiveModel
+      attr_name_override = nil
+      match = /\Acustom_field_(?<id>\d+)\z/.match(attribute)
+      if match
+        attr_name_override = CustomField.find_by_id(match[:id]).name
       end
-      full_messages
+
+      attr_name = attribute.to_s.gsub('.', '_').humanize
+      attr_name = @base.class.human_attribute_name(attribute, :default => attr_name)
+      I18n.t(:"errors.format", {
+                               :default   => "%{attribute} %{message}",
+                               :attribute => attr_name_override || attr_name,
+                               :message   => message
+                             })
     end
```

Comment:
I am aware that you are copying and then extending the method. I believe this is a good use case for [Module#prepend](http://gshutler.com/2013/04/ruby-2-module-prepend/) because you then only have to overwrite the overwriting of the attribute name. 

That would enable you to do:

´´´
module CustomFieldErrorMessages
    def full_message(attribute, message)
      match = /\Acustom_field_(?<id>\d+)\z/.match(attribute)

      if match
        attribute = CustomField.find_by_id(match[:id]).name
      end

      super
    end
end

ActiveRecord::Errors.prepend CustomFieldErrorMessages
```

At least I think this should be possible. Never tried it.